htdocs/api/v0.0.3-dev/MedhubConsent.php

```diff
@@ -0,0 +1,192 @@
+    {
+
+
+
+
+        $this->AllowedMethods = [
+            'PUT'
+        ];
+
+        $this->RequestData    = $data;
+
+        parent::__construct($method);
+    }
+
+
+
+    public function handlePUT()
+    {
+
+        $token = null;
+        $consentList = null;
+        $candid = null;
+
+        $data   = $this->RequestData;
+        if ($data === null) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("Can't parse data");
+            $this->safeExit(0);
+        }
+
+
+        if (!isset($data['Token'])) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("There is no Token object in the PUT data");
+            $this->safeExit(0);
+        }
+
+        if (!isset($data['ConsentList'])) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("There is no Consent List object in the PUT data");
+            $this->safeExit(0);
+        }
+
+        //Tries to select CandID + entry date for the given token --> later throws and error and dies if nothing found
+        $token = $data['Token'];
+        $consentList = $data['ConsentList'];
+
+        $candidTimeUsed = $this->DB->pselectRow(
+            "SELECT CandidateID, EntryDate, AlreadyUsed
+                FROM medhub_token
+             WHERE Token = :to
+                ",
+            ['to' => $token]
+        );
+
+
+        //Check that every consent is represented AND all fields are set
+        $consentNames = ['INF','HR', 'GEN', 'CL', 'CQ', 'CS', 'CR', 'ATPSY','ORPS'];
+        foreach($consentNames as $conName){
+            if (!isset($consentList[$conName]['Response'], $consentList[$conName]['Date'])){
+                error_log('Incomplete Consent Data');
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->safeExit(0);
+            }
+        }
+
+
+        if (!isset($candidTimeUsed['EntryDate'],$candidTimeUsed['CandidateID'] )or isset($candidTimeUsed['AlreadyUsed'])){
+            error_log("Error: No candidate found for token $token or token already used");
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->safeExit(0);
+        }
+
+        $entryDate = $candidTimeUsed['EntryDate'];
+        $candid = $candidTimeUsed['CandidateID'];
+
+
+
+
+        //Checks if token more than a ~month old
+        if((time()-(60*60*24*30)) > strtotime($entryDate)){
+            error_log("The Given Token is expired: we are unable to connect this to a file");
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->safeExit(0);
+        }
+
+
+
+        //TODO: Incude validation to make sure the consent object is complete (IE 9 rows), and that every row is complete!
+
+
+        //Gets the mapping of consent IDs, Names, and Labels from the DB
+        $consentIDLabel = $this->DB->pselectWithIndexKey(
+            "SELECT ConsentID, Name, Label
+                FROM consent" ,
+            [], Name
```

Comment:
```suggestion
            [], "Name"
```

you have to pass a string object

Reply:
Weirdly this has been working, but Ill change to string